config/tools/config_generator.cpp

```diff
@@ -0,0 +1,896 @@
+        return value;
+    }
+
+    [[nodiscard]] auto config_generator::get_param_from_template_file(
+        const std::string option,
+        std::map<std::string, std::string>& config_map)
+        -> std::variant<std::string, size_t, double> {
+        auto it = config_map.find(option);
+        if(it != config_map.end()) {
+            value_t parsed_val = parse_value(it->second, false);
+            if(std::holds_alternative<size_t>(parsed_val)) {
+                return std::get<size_t>(parsed_val);
+            } else if(std::holds_alternative<double>(parsed_val)) {
+                return std::get<double>(parsed_val);
+            } else if(std::holds_alternative<std::string>(parsed_val)) {
+                return std::get<std::string>(parsed_val);
+            } else {
+                __builtin_unreachable();
+            }
+        } else {
+            std::string error_msg
+                = "Warning: Could not find param, " + option + ".";
+            std::cout << error_msg << std::endl;
+            return error_msg;
+        }
+    }
+
+    void config_generator::set_param_to_config_file(const std::string key,
+                                                    const std::string value) {
+        m_new_config << key << "=" << '"' << value << '"' << '\n';
+    }
+
+    void config_generator::set_param_to_config_file(const std::string key,
+                                                    const double value) {
+        m_new_config << key << "=" << value << '\n';
+    }
+
+    void config_generator::set_param_to_config_file(const std::string key,
+                                                    const size_t value) {
+        m_new_config << key << "=" << value << '\n';
+    }
+
+    void config_generator::set_log_level(const std::string key,
+                                         std::string& log_level) {
+        if(log_levels.find(log_level) == log_levels.end()) {
+            log_level = "DEBUG";
+            std::cout << "Warning: Log level not recognized. Setting to DEBUG"
+                      << std::endl;
+        } else if(find_value<std::string>(key).has_value() == false) {
+            log_level
+                = find_value<std::string>(tmpl_universal_override_log_level)
+                      .value();
+        }
+    }
+
+    [[maybe_unused]] auto
+    config_generator::create_component(const char* type,
+                                       const value_t component_count,
+                                       bool create_2pc) -> std::string {
+        std::string return_msg = "";
+        if(std::holds_alternative<size_t>(component_count)) {
+            auto _default_log_level
+                = find_value<std::string>(tmpl_default_log_level);
+            if(std::get<size_t>(component_count) == 0) {
+                return_msg += "Warning: 0 count for at least one component. "
+                              "Fix configuration template and rerun.\n";
+            } else if(create_2pc == true) {
```

Comment:
```suggestion
            } else if(create_2pc) {
```

Reply:
Changed all these. Generally i like to explicitly do the == true or == false since it is a lot more legible than the absence or presence of "!" which can be easily missed. But i have changed all these.

DONE

Reply:
This is also a cosmetic view, but I hold it a lot more strongly.

Personally, I don't find the additional information to be more legible, and the choice to include more doesn't include an obvious stopping point to me (`create_2pc` ≡ `create_2pc == true` ≡ `create_2pc == true == true == true`, etc.). To that end, the only obvious bright-line to draw for me is “comparison operators should be used when you need to convert something into a `bool` to make a decision.” Because `create_2pc` is already a `bool`, there's no need for an additional comparison operator.